src/js/util/walkthrough-utils.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+/**
+ * Walkthrough Utility has functions used in the walkthrough react components
+ * that need to be well tested and can exist outside of the ui
+ */
+
+/**
+ * The container for the property values in the store. These differ by primitive type.
+ * @TODO update these when guide and scales are added into redux
+ * @type {Object}
+ */
+var PROPERTY_CONTAINER = {
+  marks: 'properties.update',
+  guides: 'property',
+  scales: ''
+};
+
+/**
+ * Get the properties off the primitive
+ * @param  {Object} object - a primitive object
+ * @param  {String} type - either marks, guides or scales
+ * @returns {Object} - object of the primitives properties
+ */
+function getProperties(object, type) {
+  var props = PROPERTY_CONTAINER[type].split('.');
+  var propertyObj = object;
+  for (var i = 0; i < props.length; i++) {
+    propertyObj = propertyObj ? propertyObj[props[i]] : propertyObj;
+  }
+  return propertyObj;
+}
+
+/**
+ * Get an array of all elements with a certain type
+ * @param  {string} elemType - a string that will match the key in each state element
+ * @param  {Array} stateArray - an array of objects to test against
+ * @param  {string} key - key to filter by
+ * @returns {Object}
+ */
+function filterBy(elemType, stateArray, key) {
+  return stateArray.filter(function(o) {
+    return o[key] === elemType;
+  });
+}
+
+/**
+ * Test that the type exists in the state
+ * @param  {string} elemType - a string that will match the 'type' key in each state element
+ * @param  {Array} stateArray - an array of objects to test against
+ * @param  {number} min - minimum required to be found to pass test
+ * @returns {Boolean} Success or failure of the
+ */
+function testTypeExistence(elemType, stateArray, min) {
+  var minimum = min || 1;
+  var found = filterBy(elemType, stateArray, 'type');
+
+  return (found.length >= minimum);
+}
+
+/**
+ * Test that an element exists with all properties needed
+ * @param  {Object} match - the matching element
+ * @param  {Object} elem - an object to match against
+ * @param  {String} type - the type in the store eg - Marks, Scales, Guide
+ * @returns {Boolean} Success or failure of the test
+ */
+function matchPropertyExistence(match, elem, type) {
+  // get the container of the properties by type
+  var matchProps = getProperties(match, type),
+      elemProps = getProperties(elem, type);
+  return _.isMatch(elemProps, matchProps);
```

Comment:
Still some lodash here! Is it simple enough to write ourselves, or we could [require lodash-ismatch](https://www.npmjs.com/package/lodash.ismatch)?